burn-tensor/src/tensor/ops/tensor.rs

```diff
@@ -65,6 +65,25 @@ pub trait TensorOps<B: Backend> {
         Self::from_data(Data::ones(shape), device)
     }
 
+    /// Creates a tensor filled with given value.
+    /// 
+    /// # Arguments
+    /// 
+    /// * `shape` - The shape of the tensor.
+    /// * `fill_value` - The value with which to fill the tensor.
+    /// * `device` - The device to create the tensor on.
+    /// 
+    /// # Returns
+    /// 
+    /// The tensor filled with given value
+    fn full<const D: usize>(
+        shape: Shape<D>, 
+        fill_value: B::FloatElem, 
+        device: &B::Device
+    ) -> B::TensorPrimitive<D> {
+        Self::from_data(Data::full(shape, fill_value), device)
```

Comment:
I think I would use `Self::zeros(shape, device) + fill_value` for the default implementation, it's more performant on most backends. We can also remove the `burn-wgpu` implementation since it's pretty much what it does. The implementation can be the same for `int_full`.